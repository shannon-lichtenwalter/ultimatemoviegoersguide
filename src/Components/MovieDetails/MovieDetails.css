.detailedPage {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: auto 0;
  width: 100%;
  
}

.detailedPage > div {
  width: 80%;
  display: flex;
  flex-direction: column;
  padding: 4%;
}

.image-holder {
  align-items:center;
}

.the-details-page{
  display:flex;
  flex-direction: column;
}

.back-button {
  margin: 10px auto;
  border: 1px solid lightgrey;
  border-radius: 5px;
  font-size: 1.3rem; 
  background-color: transparent;
}

.back-button:hover {
  background-color: rgb(231, 231, 227, 0.4);
}

.allMovieDetails > h2 {
  font-size: 1.5rem;
  text-align:center;
}

h3 {
  padding: 3% 0;
  font-size: 1rem;
  font-style: italic;
}

.category {
  font-weight: bold;
}

.allMovieDetails > p {
  padding: 5px 0;
}

.overview {
  width: 90%;
}
.allMovieDetails > p, .allMovieDetails > h2, .allMovieDetails > h3 {
  margin: 0;
}

.allMovieDetails {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 60%;
}

.allMovieDetails > p {
  text-align: left;
}

.similarFilms {
  text-align:center;
  margin-top: 50px;
}

.posterImage, .poster {
  -o-box-shadow:      12px 12px 29px #555;
  -icab-box-shadow:   12px 12px 29px #555;
  -khtml-box-shadow:  12px 12px 29px #555;
  -moz-box-shadow:    12px 12px 29px #555;
  -webkit-box-shadow: 12px 12px 29px #555;
  box-shadow:         12px 12px 29px #555;
}

@media screen and (min-width: 680px) {
  .allMovieDetails > h2 {
    text-align:left;
  }

  .detailedPage{
    flex-direction: row;
    justify-content: center;
  }
  .detailedPage > .image-holder {
    align-items: flex-end;
    width: 50%;
  }

  .detailedPage > .allMovieDetails {
    align-items: flex-start;
    width: 50%;
  }

}

@media screen and (min-width: 900px) {
  .overview {
    width: 60%;
  }
}