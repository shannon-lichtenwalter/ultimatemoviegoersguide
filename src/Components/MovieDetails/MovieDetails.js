import React from 'react';
import { withRouter } from 'react-router-dom';
import GetMovieLists from '../../Services/getMovieLists'
import Movie from '../Movie/Movie';
import './MovieDetails.css';

class MovieDetails extends React.Component {
  state = {
    movieData: null,
    similarFilms: null
  }

  naughtyWordEraser = (string) => {
    const naughtyWordBank = ['fuck', 'fucking', 'shit', 'damn', 'damnit', 'ass'];
    let array= string.split(' ');
    let badWord=false;

    let cleanArray = array.map(word => {
      if(naughtyWordBank.includes(word.toLowerCase())){
        badWord=true;
        let filteredWord = word[0];
        for(let i=1; i<word.length; i++){
          filteredWord += '*';
        }
        return filteredWord;
      } else {
        return word
      }
    })
    
    if(!badWord){
      return string;
    }

    let newString= '';
    cleanArray.map(word => newString += `${word} `);
    return newString;
  }

  listGenres = (array) => {
    let string = '';
    for (let i = 0; i < array.length; i++) {
      if(array.length === 1){
        string += array[i].name
      }

      else if (array.length === 2){
        if(i=== 0){
          string += array[i].name
        } else {
          string += ' & ' + array[i].name
        }
      }

      else if (i === array.length - 1) {
        string += '& ' + array[i].name
      } else {
        string += array[i].name + ', '
      }
    }
    return string;
  }

  componentDidMount = () => {
    document.body.scrollTop = 0;
    document.documentElement.scrollTop = 0;

    GetMovieLists.getMovie(this.props.match.params.id)
      .then((res) => {
        this.setState({
          movieData: res
        })
      }).catch(e => this.props.setError(e))

    GetMovieLists.getSimilarFilms(this.props.match.params.id)
      .then((res) => {
        this.setState({
          similarFilms: res
        })
      }).catch(e => this.props.setError(e))
  }

  render() {
    return (
      <section>
        {!this.state.movieData && <h3 className='loading'>Loading...</h3>}
        {this.state.movieData && <div className='the-details-page'>
          <div className='detailedPage'>
            <div className='image-holder'>
              {this.state.movieData.poster_path
                ? <img className='posterImage' src={`https://image.tmdb.org/t/p/w300/${this.state.movieData.poster_path}`} alt='movie poster' />
                : <div className='poster'>No Poster Data Available</div>
              }
            </div>
            <div className='allMovieDetails'>
              <h2>{this.state.movieData.title}</h2>
              <h3>{this.naughtyWordEraser(this.state.movieData.tagline)}</h3>
              <p><span className='category'>Release Date:</span> {new Date(this.state.movieData.release_date).toLocaleDateString()}</p>
              <p><span className='category'>Genres:</span> {this.listGenres(this.state.movieData.genres)}</p>
              <p><span className='category'>Average Votes:</span> {this.state.movieData.vote_average}/10</p>
              <p><span className='category'>Original Title:</span> {this.state.movieData.original_title}</p>
              <p><span className='category'>Runtime:</span> {this.state.movieData.runtime} minutes</p>
              <p className='overview'>{this.state.movieData.overview}</p>
            </div>
          </div>
          <button className='back-button' onClick={() => this.props.history.goBack()}>Go Back</button>
          <h2 className='similarFilms'>Similar Movies:</h2>
            <ul className='movies'>
            {!this.state.similarFilms && <li>...searching</li>}
            {this.state.similarFilms && this.state.similarFilms.results.length === 0 && <li>No Similar Films Found In Database...</li>}
            {this.state.similarFilms && this.state.similarFilms.results.map((movie, index) => {
              return <Movie key={index} data={movie} showDetails={this.props.showDetails} />
            })}
          </ul>
        </div>
        }
      </section>
    )
  }
}

export default withRouter(MovieDetails);