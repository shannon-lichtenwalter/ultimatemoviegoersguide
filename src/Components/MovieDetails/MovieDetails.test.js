import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MovieDetails from './MovieDetails';

describe('MovieDetails Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter><MovieDetails /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  }); 

  it('Displays the MovieDetails component correctly when rendered', () => {
    const wrapper = mount(<BrowserRouter><MovieDetails /></BrowserRouter>);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});