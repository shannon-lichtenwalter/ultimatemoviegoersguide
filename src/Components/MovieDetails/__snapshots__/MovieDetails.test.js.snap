// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Component Displays the MovieDetails component correctly when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(MovieDetails)>
      <MovieDetails
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <section>
          <h3
            className="loading"
          >
            Loading...
          </h3>
        </section>
      </MovieDetails>
    </withRouter(MovieDetails)>
  </Router>
</BrowserRouter>
`;
